### ISSUE TITLE
Support long presses as keymaps.

### ISSUE THREAD

**Issue by SirVer** [swiboe/swiboe#10]
On reddit somebody yesterday suggested to support long presses for keymaps - which is a great idea imho. 

I am not sure if rustbox supports keyup events - otherwise we'd need to add that to it.

**Comment by gchp** [comment]
Hey @SirVer! I'm the maintainer of rustbox. Unfortunately, it does not support keyup events. It is essentially a wrapper of [termbox](https://github.com/nsf/termbox), which itself does not support keyup events.

Might be worth investigating, but not sure if its actually possible :/

**Comment by gchp** [comment]
Really liking this project BTW, very nice! I've been playing around with a [similar project](https://github.com/gchp/iota) myself, and I might borrow some of your ideas :)

**Comment by SirVer** [comment]
@gchp Thanks for the comments! looked at iota but think it's design is too similar to Vim regarding extensibility. Swiboe is build around the concept of extensibility. 

I really want to make Swiboe into one of the big text editors and could use any support - and of course more than one persons vision. I think a lot of stuff you implemented for iota could be made into a plugin for Swiboe - the other way around will probably not work easily.

I dug a bit and learned that terminals do not surface that sort of information - so key up events are not represented there. That means it is not rustbox that simply did not surface it. See https://github.com/nsf/termbox-go/issues/83. I think it can be emulated with listening in to key-repeat events - but that is clunky. And of course a GUI could surface that information - so the keymap handler should be able to deal with it.

**Comment by gchp** [comment]
Yeah, I hadn't really gotten to thinking about extensibility just yet. I'm really just playing around with lots of different ideas to see what works well. I like how Swiboe has done things, and I'd like to play around with some of the plugin stuff. I agree with you though, as it is now iota is not built for extensibility! I plan on messing with Swiboe this weekend, so I may send a few PRs your way! Keep up the good work!

**Comment by SirVer** [comment]
Please be gentle when you looks through the code. Swiboe is in 'outline mode' - i.e. I add features without much error checking and little testing to see if the big design that I have in mind works. There are a couple of things that are not yet clear how they will fit together and I need to test them out before committing. So far everything looks good - I think the last remaining big thing before cleaning up is clients in other languages - I'll work on a python client on stream next, I think.

feel free to poke me here or on twitter if you need anything. If you want to, we could also remote pair using floobits or simple video chat. That will probably get you started the quickest.

**Comment by gchp** [comment]
@SirVer it would be awesome to pair on this sometime! Maybe in the next few weeks we could arrange something?

Excited to see the Python client. Python is probably my "main" language other than Rust at the moment, so having that would be great. Happy to help on that too if needs be!

**Comment by SirVer** [comment]
I started on the python client on stream. Just check by at 8am CEST on http://twitch.tv/sirverii and say hi. I want to keep the streams consistent, so that viewers can follow along daily, so we cannot work on that off stream. Either we pair on stream - which would be really cool but also scary from a technical point of view or we pair on something else.

I am always up for pairing - either on the weekends, early morning or nights. I hang out in the swiboe gitter chat when I am online, you can poke me there and we can fix a date.
